lib/adagrams.rb

```diff
@@ -0,0 +1,136 @@
+require 'pry'
+
+#create data structure
+all_letters = []
+
+letters = {"A" => 9, "B" => 2, "C" => 2, "D"=> 4, "E" => 12, "F"=> 2, "G"=>3, "H"=>2, "I" => 9, "J"=>1, "K"=> 1, "L"=>4, "M"=> 2, "N"=> 6, "O"=>8, "P" => 2, "Q" => 1, "R"=>6, "S"=>4, "T"=>6, "U"=>4, "V"=>2, "W"=>2, "X"=>1, "Y"=>2, "Z"=>1}
+
+
+# Fill pool of letters array with letters
+letters.each do |letter, freq|
+  count = 0
+  until count == freq
+    all_letters << letter
+    count += 1
+  end
+end
+
+
+# Define method for creating user hand
+def draw_letters(array)
+  user_letters = array.sample(10)
+  return user_letters
+end
+
+DRAWN_LETTERS = draw_letters(all_letters)
+
+
+# Transform input word string into array, all in capital letters
+def make_word_array(input)
+  input = input.upcase.split("")
+  return input
+end
+
+
+#create method to see if word uses letters that are actually available
+def uses_available_letters? (input, drawn_letters)
+  is_valid = "default"
+  word_letters = make_word_array(input)
+  word_letters.each do |letter|
+    if drawn_letters.include?letter
+      is_valid = true
+      puts "letter valid " + "#{is_valid}"
+      letter_index = drawn_letters.index(letter)
+      drawn_letters.delete_at(letter_index)
+    else
+      is_valid = false
+      puts "letter not valid " + "#{is_valid}"
+      return is_valid
+    end
+  end
+  return is_valid
+end
+
+
+puts "drawn letters: #{DRAWN_LETTERS}"
+
+
+#turn all_letters array into a hash that represents the score of each letter
+LETTER_SCORES = {}
+
+# Make hash for letter to score
+all_letters.uniq.each do |letter|
+
+  case letter
+  when "A","E", "I", "O", "U", "L", "N", "R", "S", "T"
```

Comment:
The idea of generating this data structure dynamically with a loop and a case statement is an interesting one. However, I think I would argue that the resulting code is a little less readable than just hardcoding it:

```ruby
LETTER_SCORES = {
  "A" => 1,
  "B" => 3,
  # ...
}
```